libs/perun/dialogs/src/lib/mail-change-failed-dialog/mail-change-failed-dialog.component.ts

```diff
@@ -0,0 +1,22 @@
+import { Component, OnInit } from '@angular/core';
+import { MatDialogRef } from '@angular/material/dialog';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'perun-web-apps-mail-change-failed-dialog',
+  templateUrl: './mail-change-failed-dialog.component.html',
+  styleUrls: ['./mail-change-failed-dialog.component.scss']
+})
+export class MailChangeFailedDialogComponent implements OnInit {
+
+  constructor(private dialogRef: MatDialogRef<MailChangeFailedDialogComponent>,
+              private router: Router) { }
+
+  ngOnInit(): void {
+  }
+
+  onClose() {
+    this.router.navigate([], {queryParamsHandling: ''})
```

Comment:
I'm not sure if you want to always redirect to `/`. I think you should navigate to the same path, where the user already is, only changes the parameters. Also, the `queryParamsHandling` parameter has only two valid options: 'merge' and 'preserve'. When you use `''` I think it's the same as not specifying it at all?

Reply:
Since the link that you get in the confirmation email points to / and logic for handling pref mail change is implemented on home pages for both user-profile and admin-gui, there is no reason to redirect elsewhere. As for the queryParamsHandling, you are right, it can be omitted.

Reply:
Ok, we can keep it this way.